glusterd2/config.go

```diff
@@ -60,6 +60,9 @@ func parseFlags() {
 	flag.String("cert-file", "", "Certificate used for SSL/TLS connections from clients to glusterd2.")
 	flag.String("key-file", "", "Private key for the SSL/TLS certificate.")
 
+	// PID file
+	flag.String("pidfile", "", "PID file path")
```

Comment:
I don't think so. Most of the daemons provide option to customize path. (`grep -R "\-\-pid" /usr/lib/systemd/system/`). `sshd` supports the same via config file(`sshd_config`)

All Gluster daemons support configuring pid file, including Glusterd1.

Glusterd1 tests uses custom pid file for each spawned glusterd, these paths should be predictable so that it can be signaled. UUID is good idea to differentiate between multiple glusterds, but not predictable.

Reply:
> these paths should be predictable

If the usecase is limited to tests, we get predictability from explicitly setting `rundir`. And also implicitly by `workdir` when`rundir` isn't specified..

Example as per current e2e paths:

`/tmp/gd2_func_test/<workdir>/run/gluster/glusterd2.pid`

Reply:
In any case, at least during graceful shutdowns, glusterd2 should delete the pidfile it created (in signal handler)